src/RequestHandle.cpp

```diff
@@ -480,4 +562,83 @@ auto curl_write_data(
     return data_length;
 }
 
+auto curl_xfer_info_old(
+    void* pointer_to_request,
+    double, double , double , double
+) -> int
+{
+    auto* raw_request_ptr = static_cast<RequestHandle*>(pointer_to_request);
+
+    /**
+     * If m_bytes_written is greater than or equal to m_max_download_bytes,
+     * we can terminate this request.
+     * Returning 1 will cause the cURL request to be aborted.
+     * (Note: This will result in the CURLcode returning as an error.
+     *  This is handled in setCompletionStatus.)
+     **/
+    if (raw_request_ptr->getBytesLeftToWrite() == 0)
+    {
+        return 1;
+    }
+
+    // Returning 0 will result in cURL request continuing normally.
+    return 0;
+}
+
+auto curl_xfer_info_new(
+    void* pointer_to_request,
+    curl_off_t, curl_off_t, curl_off_t, curl_off_t
+) -> int
+{
+    auto* raw_request_ptr = static_cast<RequestHandle*>(pointer_to_request);
+
```

Comment:
return raw_request_ptr->hasDownloadedRequiredBytes() -> int;  This function is duplicated with its sister function

Reply:
Realized today that we don't need these callbacks to abort the CURL request, so I've removed them. More in main comment.